scipy/signal/_signaltools.py

```diff
@@ -2476,19 +2476,23 @@ def deconvolve(signal, divisor):
 def hilbert(x, N=None, axis=-1):
     r"""FFT-based computation of the analytic signal.
 
-    The analytic signal is calculated by filtering out the negative frequencies and
+    The analytic signal is calculated by zeroing out the negative frequencies and
     doubling the amplitudes of the positive frequencies in the FFT domain.
     The imaginary part of the result is the hilbert transform of the real-valued input
     signal.
     
     The transformation is done along the last axis by default.
 
+    For numpy arrays, `scipy.fft.set_workers` can be used to change the number of
+    workers used for the FFTs.
+
     Parameters
     ----------
     x : array_like
         Signal data.  Must be real.
     N : int, optional
-        Number of Fourier components.  Default: ``x.shape[axis]``
+        Number of Fourier components. ``x`` is cropped or zero-padded to length
+        ``N`` along ``axis``.  Default: ``x.shape[axis]``
```

Comment:
A cosmetic remark: I think denoting `N` as number of Fourier components is not precise, since `x_a` has at most ``N//2 + 1`` non-zero Fourier components. Furthermore, the FFT is shortened or zero-padded.
What about something like:
```suggestion
        Number of output samples. The output length is adjusted by cropping or or zero-padding
        the FFT of `x` accordingly. Default: ``x.shape[axis]``
```

PS: In docstrs function parameters are enclosed by single ticks, e.g., `` `x` `` and non-parameters and code by two single ticks, e.g., ``` ``x.shape[axis]`` ```.

Reply:
Agreed on the original version being imprecise. I updated to something similar to your suggestion, although I believe `fft()` pads the original signal, not the fourier coefficients.
[
Docs](https://output.circle-artifacts.com/output/job/870fee4e-e67a-416e-b886-527dbab091f0/artifacts/0/html/reference/generated/scipy.signal.hilbert.html) seem to render/link correctly.

Reply:
I like you're corrected version.